corehq/messaging/scheduling/scheduling_partitioned/models.py

```diff
@@ -634,10 +635,14 @@ def recipient(self):
                 settings.AVAILABLE_CUSTOM_SCHEDULING_RECIPIENTS[self.recipient_id][0]
             )
             return custom_function(self)
-        elif self.recipient_type == self.RECIPIENT_TYPE_CASE_PROPERTY_USER:
+        elif self.recipient_type == self.RECIPIENT_TYPE_CASE_PROPERTY_USERNAME:
             username = self.case.get_case_property(self.recipient_id)
-            full_username = format_username(username, self.domain)
-            return CommCareUser.get_by_username(full_username)
+            if username:
```

Comment:
nit: This check is redundant since `get_by_username` already returns None if `username` is Falsy

Reply:
`format_username` also handles that via `username or ''`, but it's fine with me if you prefer to do the check